src/Stratis.Bitcoin/Controllers/RestApiClientBase.cs

```diff
@@ -118,9 +118,103 @@ public RestApiClientBase(ILoggerFactory loggerFactory, IFederationGatewaySetting
             return null;
         }
 
+        public async Task<Response> SendGetRequestAsync<Response>(string apiMethodName, string arguments = null,
+            CancellationToken cancellation = default(CancellationToken)) where Response : class
+        {
+            HttpResponseMessage response = await this.SendGetRequestAsync(apiMethodName, arguments, cancellation).ConfigureAwait(false);
+
+            // Parse response.
+            if ((response != null) && response.IsSuccessStatusCode && (response.Content != null))
+            {
+                string successJson = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                if (successJson != null)
+                {
+                    Response responseModel = JsonConvert.DeserializeObject<Response>(successJson);
```

Comment:
If we can't de-serialize the Json we might want to catch the exception and log it.

Reply:
`JsonConvert.DeserializeObject` doesn't throw